### PR TITLE
User profiles

### LINKED ISSUE #14: Profile page
- [x] Page styling
- [x] Edit modal styling
- [x] Edit modal validation
- [x] #15 User endpoints

### LINKED ISSUE #15: Create user endpoints
Just `PATCH` for now

### LINKED ISSUE #16: Create reusable tags
- [x] Tag component and styling 
- [x] Self-contained interests server side with a reference in user model

### PR DESCRIPTION
**Pull request by MichielLeyman**
Closes #14 
Closes #15

### CODE REVIEW

**Comment by MichielLeyman** [comment]
Partly tackles #16

File: server/src/controller/user.controller.js [hunk 1]
```diff
@@ -10,6 +10,17 @@ exports.findAll = (req, res) => {
     });
 };
 
+exports.findOne = (req, res) => {
+  const { id } = req.params;
+  User.findById(id)
+    .select('-password')
+    .select('-passwordResetToken')
```
Comment (Toinne) [inline server/src/controller/user.controller.js]: PP: would add some middleware https://mongoosejs.com/docs/middleware.html as this is easy to forget and dangerous to forget

File: server/src/model/user.model.js [hunk 2]
```diff
@@ -10,22 +10,38 @@ const userSchema = new mongoose.Schema(
       type: String,
       required: true
     },
-    about: {
-      type: String,
-      required: true
+    bio: {
```
Comment (Toinne) [inline server/src/model/user.model.js]: REQ: seed script changes required

File: client/src/components/Pages/Profile/ProfileForm.jsx [hunk 3]
```diff
@@ -0,0 +1,200 @@
...
+  TextInputField,
+  Button,
+  TagInput
+} from 'evergreen-ui';
+import {
+  API,
+  IS_LOADING,
+  INACTIVE,
+  HAS_ERRORED
+} from '../../../constants';
```
Comment (Toinne) [inline client/src/components/Pages/Profile/ProfileForm.jsx]: PP: This component is getting pretty big would pull the api interaction up one level.

File: client/src/api/API.js [hunk 4]
```diff
@@ -0,0 +1,33 @@
+import { API_URL } from '../constants';
+
+class API {
+  constructor() {
+    this.BASE_URL = API_URL;
+    this.headers = {
```
[the remark this answers is not in the document]
Comment (MichielLeyman) [reply client/src/api/API.js]: This'll allow me to selectively add and remove an Authorization header down the line :)

File: server/src/controller/user.controller.js (hunk 1 above)
Comment (MichielLeyman) [reply server/src/controller/user.controller.js]: Removed and commented in https://github.com/HackYourFutureBEHomework/class2-project-bementor/pull/18/commits/4256907bb8752e4149d3dbc538cd2acf9f9e0964

File: client/src/components/Pages/Profile/ProfileForm.jsx (hunk 3 above)
Comment (MichielLeyman) [reply client/src/components/Pages/Profile/ProfileForm.jsx]: Tackled in #20

**Pull request merged**